### PR TITLE
System tests

### PR DESCRIPTION
**Pull request by ivankocienski**
Is solving #870 where we use browser testing to verify the front end is running. So far only the system test framework has been set up and an example test has been written. Later tests will deal specifically with the javascript components.

### CODE REVIEW

**Comment by aaaaargZombies** [comment]
I'm not sure I'm qualified to review the ruby, so maybe this won't move you closer to merging but a couple of notes.

I was able to run the rails app as normal and the tests via `rake test:system`. I did receive a warning.

```
2022-07-12 09:56:20 WARN Selenium [DEPRECATION] [:browser_options] :options as a parameter for driver initialization is deprecated. Use :capabilities with an Array of value capabilities/options if necessary instead.
Capybara starting Puma...
* Version 5.6.4 , codename: Birdie's Version
* Min threads: 0, max threads: 4
* Listening on http://127.0.0.1:38159
.

Finished tests in 28.603325s, 0.0350 tests/s, 0.1748 assertions/s.


1 tests, 5 assertions, 0 failures, 0 errors, 0 skips
```

Maybe this is expected.

I think it would be useful to have some notes about how to run tests and how to add tests now it's been set up (I'm guessing rails is like node and there are many options). Maybe in the readme or perhaps the wiki.

**Comment by ivankocienski** [comment]
> I'm not sure I'm qualified to review the ruby, so maybe this won't move you closer to merging but a couple of notes.
> 
> I was able to run the rails app as normal and the tests via `rake test:system`. I did receive a warning.
> 
> ```
> 2022-07-12 09:56:20 WARN Selenium [DEPRECATION] [:browser_options] :options as a parameter for driver initialization is deprecated. Use :capabilities with an Array of value capabilities/options if necessary instead.
> Capybara starting Puma...
> * Version 5.6.4 , codename: Birdie's Version
> * Min threads: 0, max threads: 4
> * Listening on http://127.0.0.1:38159
> .
> 
> Finished tests in 28.603325s, 0.0350 tests/s, 0.1748 assertions/s.
> 
> 
> 1 tests, 5 assertions, 0 failures, 0 errors, 0 skips
> ```
> 
> Maybe this is expected.
> 
> I think it would be useful to have some notes about how to run tests and how to add tests now it's been set up (I'm guessing rails is like node and there are many options). Maybe in the readme or perhaps the wiki.

The warnings are there (for me as well) and I should really fix them, good catch.

I can update the README with testing instructions!

**Comment by aaaaargZombies** [comment]
The info in the readme is a big improvement :+1:

**Review by alexandria-gfsc — APPROVED** [review]
lgtm

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -51,7 +66,7 @@ jobs:
       - name: Do Tests
         run: |
           yarn build
-          bundle exec rake
+          bundle exec rake && bundle exec rake test:system
```
[the remark this answers is not in the document]
Comment (ivankocienski) [reply .github/workflows/test.yml:69]: I assume it has the same functionality of conditional execution based on the result of the previous step?

**Pull request merged**